plugin/advisor/mysql/mysql_where_requirement_advisor.go

```diff
@@ -0,0 +1,89 @@
+package mysql
+
+import (
+	"fmt"
+
+	"github.com/bytebase/bytebase/common"
+	"github.com/bytebase/bytebase/plugin/advisor"
+	"github.com/bytebase/bytebase/plugin/db"
+
+	"github.com/pingcap/tidb/parser/ast"
+)
+
+var (
+	_ advisor.Advisor = (*WhereRequirementAdvisor)(nil)
+)
+
+func init() {
+	advisor.Register(db.MySQL, advisor.MySQLWhereRequirement, &WhereRequirementAdvisor{})
+	advisor.Register(db.TiDB, advisor.MySQLWhereRequirement, &WhereRequirementAdvisor{})
+}
+
+// WhereRequirementAdvisor is the advisor checking for the WHERE clause requirement for UPDATE/DELETE.
+type WhereRequirementAdvisor struct {
+}
+
+// Check checks for the WHERE clause requirement for UPDATE/DELETE.
+func (adv *WhereRequirementAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	p := newParser()
+
+	root, _, err := p.Parse(statement, ctx.Charset, ctx.Collation)
+	if err != nil {
+		return []advisor.Advice{
+			{
+				Status:  advisor.Error,
+				Title:   "Syntax error",
+				Content: err.Error(),
+			},
+		}, nil
+	}
+
+	we := &whereRequirementChecker{}
+	for _, stmtNode := range root {
+		(stmtNode).Accept(we)
+	}
+
+	if len(we.advisorList) == 0 {
+		we.advisorList = append(we.advisorList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    common.Ok,
+			Title:   "OK",
+			Content: "DELETE/UPDATE statements contain WHERE clause",
```

Comment:
I guess I would just use empty string ""

Reply:
BTW, though SELECT won't use this path for now, it may happen in the future, so I would be less explicit about the query type.